src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -440,11 +441,17 @@ private static boolean isAloneOnLine(Details details, String targetSrcLine) {
      * @return if the double brace initialization rcurly should be skipped over by the check
      */
     private static boolean skipDoubleBraceInstInit(Details details) {
-        final DetailAST rcurly = details.rcurly;
+        boolean skipDoubleBraceInstInit = true;
         final DetailAST tokenAfterNextToken = Details.getNextToken(details.nextToken);
-        return rcurly.getParent().getParent().getType() == TokenTypes.INSTANCE_INIT
-                && details.nextToken.getType() == TokenTypes.RCURLY
-                && rcurly.getLineNo() != Details.getNextToken(tokenAfterNextToken).getLineNo();
+        if (tokenAfterNextToken != null) {
+            final DetailAST rcurly = details.rcurly;
+            skipDoubleBraceInstInit = rcurly.getParent().getParent()
+                    .getType() == TokenTypes.INSTANCE_INIT
+                    && details.nextToken.getType() == TokenTypes.RCURLY
+                    && rcurly.getLineNo() != Details.getNextToken(tokenAfterNextToken)
```

Comment:
can be replaced with TokenUtil.areonSameLine

Reply:
Done